distribution/tools/server-cli/src/test/java/org/elasticsearch/server/cli/ServerProcessTests.java

```diff
@@ -246,18 +255,12 @@ public void testBootstrapError() throws Exception {
         assertThat(terminal.getErrorOutput(), containsString("a bootstrap exception"));
     }
 
-    public void testStartError() throws Exception {
+    public void testStartError() {
         processValidator = pb -> { throw new IOException("something went wrong"); };
-        var e = expectThrows(UncheckedIOException.class, () -> runForeground());
+        var e = expectThrows(UncheckedIOException.class, this::runForeground);
         assertThat(e.getCause().getMessage(), equalTo("something went wrong"));
     }
 
-    public void testOptionsBuildingInterrupted() throws Exception {
```

Comment:
Why is this test removed?

Reply:
This test was a bit "artificial": it showed that when OptionsBuilder was throwing inside of `ServerProcess#start`, the exception was propagated. We do not call `OptionsBuilder` inside start anymore, we call it outside and pass the resulting list of jvm options to the builder. We completely removed the OptionsBuilder in the end, so I think this is not needed anymore.

Reply:
Could this test be moved to JvmOptionsParserTests then?

Reply:
Not really; error handling from JvmOptionsParser is already covered by tests in JvmOptionsParserTests. Since we extracted the method call to an upper level, the equivalent test would be to make JvmOptionsParser (or something mocking it) throw inside `ServerCli#startServer`; I'll add a test in this direction to ServerCliTests.